yunhanw-google/connectedhomeip
# [im/subscribe] Enable subscription over interaction model protocol

#### Problem
The current "report" routine is not spec compliance.

#### Change overview

- Update templates to use interaction model APIs for sending subscribe
requests.
- Update DeviceControllerInteractionModelDelegate to handle report
messages from subscription.
- Update ChipCallback Mgr to bridge subscribe responses to existing
callbacks, and handle TLV message just as read responses.


#### Testing

- Update TestSuite for basic Subscription protocol tests, test routing
looks like follows:
  - Send subscribe request, wait for response.
  - Wait for a few seconds, and execute "kick" commands.
  - The test is expected to receive the same number of Reports as the
  number of kick commands.
  - The test will fail when not receiving enough report data from server
  in global timeout.
- Update python script test to (in cirque) include a similar test
routine as test suite.
